SmartDeviceLink_Example/MenuManager.swift

```diff
@@ -0,0 +1,193 @@
+            showPerformInteractionChoiceSet(with: manager)
+        })
+    }
+
+    /// Menu item that starts recording sounds via the in-car microphone when selected
+    ///
+    /// - Parameter manager: The SDL Manager
+    /// - Returns: A SDLMenuCell object
+    class func menuCellRecordInCarMicrophoneAudio(with manager: SDLManager) -> SDLMenuCell {
+        if #available(iOS 10.0, *) {
+            let audioManager = AudioManager(sdlManager: manager)
+            return SDLMenuCell(title: ACRecordInCarMicrophoneAudioMenuName, icon: SDLArtwork(image: UIImage(named: MicrophoneBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [ACRecordInCarMicrophoneAudioMenuName], handler: { (triggerSource) in
+                audioManager.startRecording()
+            })
+        }
+
+        return SDLMenuCell(title: ACRecordInCarMicrophoneAudioMenuName, icon: SDLArtwork(image: UIImage(named: SpeakBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [ACRecordInCarMicrophoneAudioMenuName], handler: { (triggerSource) in
+            manager.send(AlertManager.alertWithMessageAndCloseButton("Speech recognition feature only available on iOS 10+"))
+        })
+    }
+
+    /// Menu item that dials a phone number when selected
+    ///
+    /// - Parameter manager: The SDL Manager
+    /// - Returns: A SDLMenuCell object
+    class func menuCellDialNumber(with manager: SDLManager) -> SDLMenuCell {
+        return SDLMenuCell(title: ACDialPhoneNumberMenuName, icon: SDLArtwork(image: UIImage(named: PhoneBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [ACDialPhoneNumberMenuName], handler: { (triggerSource) in
+            guard RPCPermissionsManager.isDialNumberRPCAllowed(with: manager) else {
+                manager.send(AlertManager.alertWithMessageAndCloseButton("This app does not have the required permissions to dial a number"))
+                return
+            }
+
+            VehicleDataManager.checkPhoneCallCapability(manager: manager, phoneNumber:"555-555-5555")
+        })
+    }
+
+    /// Menu item that opens a submenu when selected
+    ///
+    /// - Parameter manager: The SDL Manager
+    /// - Returns: A SDLMenuCell object
+    class func menuCellWithSubmenu(with manager: SDLManager) -> SDLMenuCell {
+        var submenuItems = [SDLMenuCell]()
+        for i in 0..<75 {
+            let submenuTitle = "Submenu Item \(i)"
+            submenuItems.append(SDLMenuCell(title: submenuTitle, icon: SDLArtwork(image: UIImage(named: MenuBWIconImageName)!, persistent: true, as: .PNG), voiceCommands: [submenuTitle, "Item \(i)", "\(i)"], handler: { (triggerSource) in
+                manager.send(AlertManager.alertWithMessageAndCloseButton("Submenu item \(i) selected!"))
+            }))
+        }
+
+        return SDLMenuCell(title: "Submenu", subCells: submenuItems)
+    }
+}
+
+// MARK: - Perform Interaction Choice Set Menu
```

Comment:
I think the perform interaction menu should be a separate class / file

Reply:
Fixed. Added PICS menu manager to both Obj-C and Swift Ex. apps